tests/test_msg_nano_signtx.py

```diff
@@ -0,0 +1,310 @@
+from keepkeylib.tools import parse_path
+from keepkeylib import messages_nano_pb2 as proto
+from keepkeylib import nano
+
+NANO_ACCOUNT_0_PATH = parse_path("m/44'/165'/0'")
+NANO_ACCOUNT_0_ADDRESS = 'xrb_1bhbsc9yuh15anq3owu1izw1nk7bhhqefrkhfo954fyt8dk1q911buk1kk4c'
+NANO_ACCOUNT_1_PATH = parse_path("m/44'/165'/1'")
+NANO_ACCOUNT_1_ADDRESS = 'xrb_3p9ws1t6nx7r5xunf7khtbzqwa9ncjte9fmiy59eiyjkkfds6z5zgpom1cxs'
+NANO_ACCOUNT_1_PUBLICKEY = 'd8fcc8344a74b81f7746964fd27f7e20f45474c3b670f0cec87a329357927c7f'
+OTHER_ACCOUNT_3_PATH = parse_path("m/44'/100'/3'")
+OTHER_ACCOUNT_3_ADDRESS = 'xrb_1x9k73o5icut7pr8khu9xcgtbaau6z6fh5fxxb5m9s3fpzuoe6aio9xjz4et'
+OTHER_ACCOUNT_3_PUBLICKEY = '74f2286a382b7a2db0693f67ea9da4a11b27c8d78dbdea4733e42db7f7561110'
+REP_OFFICIAL_1 = 'xrb_3arg3asgtigae3xckabaaewkx3bzsh7nwz7jkmjos79ihyaxwphhm6qgjps4'
+REP_NANODE = 'xrb_1nanode8ngaakzbck8smq6ru9bethqwyehomf79sae1k7xd47dkidjqzffeg'
+RECIPIENT_DONATIONS = 'xrb_3wm37qz19zhei7nzscjcopbrbnnachs4p1gnwo5oroi3qonw6inwgoeuufdp'
+RECIPIENT_DONATIONS_PUBLICKEY = 'f2612dfe03fdec8169fcaa2aad9384d28853f22b01d4e5475c5601bd69c2429c'
+
+class TestMsgNanoSignTx(common.KeepKeyTest):
+
+    def test_encode_balance(self):
+        self.assertEqual(nano.encode_balance(0), '00000000000000000000000000000000'.decode('hex'))
+        self.assertEqual(nano.encode_balance(4440329590121742105910495447534801366), '03572d26b8163ca8016a76280cb011d6'.decode('hex'))
+        self.assertEqual(nano.encode_balance(340282366920938463463374607431768211455), 'ffffffffffffffffffffffffffffffff'.decode('hex'))
+
+    def test_block_1(self):
+        self.setup_mnemonic_nopin_nopassphrase()
+        res = self.client.nano_sign_tx(
+            'Nano', NANO_ACCOUNT_0_PATH,
+            link_hash='2e3249f1ffc09608d369e01a701bf03bd05509fab262086a59d09994d315e840'.decode('hex'),
+            representative=REP_OFFICIAL_1,
+            balance=9624176000000000000000000000000,
+        )
+        self.assertIsInstance(res, proto.NanoSignedTx)
+        self.assertEqual(res.block_hash, '517565abb71bdccf03754421b1bcaee8327cfce7a571a844ae5392e851531ece'.decode('hex'))
+        self.assertEqual(res.signature, 'e5449bd97ffcd555b435f1289ebf28be4ec4bc58915f7fbdda7283f6727b134b8fb1cdde4eafcb166f2a8a6642000ef3a088661d4100cd819d8bdddb64dd3a00'.decode('hex'))
+
+    def test_block_2(self):
+        self.setup_mnemonic_nopin_nopassphrase()
+        res = self.client.nano_sign_tx(
+            'Nano', NANO_ACCOUNT_0_PATH,
+            parent_link='2e3249f1ffc09608d369e01a701bf03bd05509fab262086a59d09994d315e840'.decode('hex'),
+            parent_representative=REP_OFFICIAL_1,
+            parent_balance=9624176000000000000000000000000,
+            representative=REP_NANODE,
+            balance=9624176000000000000000000000000,
+        )
+        self.assertIsInstance(res, proto.NanoSignedTx)
+        self.assertEqual(res.block_hash, '1a3ec7d5d246aa987d99fde40ff3cadb8833941391611ec9125014d7458ac406'.decode('hex'))
```

Comment:
```suggestion
        self.assertEqual(binascii.hexlify(res.block_hash), '1a3ec7d5d246aa987d99fde40ff3cadb8833941391611ec9125014d7458ac406')
```

is preferable for these kinds of asserts, since it makes the failures easier to read.

Reply:
Resolved with 150513d